hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/SingleTransactionRecordBuilder.java

```diff
@@ -252,11 +253,19 @@ public SingleTransactionRecordBuilder automaticTokenAssociations(
         return this;
     }
 
-    public SingleTransactionRecordBuilder parentConsensusTimestamp(Timestamp parentConsensusTimestamp) {
-        this.parentConsensusTimestamp = parentConsensusTimestamp;
+    @NonNull
+    public SingleTransactionRecordBuilder parentConsensusTimestamp(@NonNull Instant parentConsensusTimestamp) {
```

Comment:
Is this method used anywhere? Otherwise, we should remove it and make `parentConsensusTime` immutable.

Reply:
Yes, it's used to set the `parentConsensusTimestamp` for a child record [here](https://github.com/hashgraph/hedera-services/blob/06746-set-generic-record-data/hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/RecordListBuilder.java#L139).

Reply:
That is also right after initialization. We can add a second constructor that takes two arguments: `consensusNow` and `parentConsensusNow`.

Reply:
Updated.